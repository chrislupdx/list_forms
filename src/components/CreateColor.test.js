import React from 'react';
import { shallow } from 'enzyme';
import CreateColor from './CreateColor';

describe('createcolor component', () => {
  it('renders a createcolor', () => {
    const wrapper = shallow(<CreateColor addColor={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('updates name on change', () => {
    const wrapper = shallow(<CreateColor addColor={jest.fn()} />);
    wrapper.find('[name="name"]').at(0).simulate('change', { target: { name: 'name', value: 'Red' }
    });
    expect(wrapper.state('name')).toEqual('Red');
  });

  it('updates color on change', () => {
    const wrapper = shallow(<CreateColor addColor={jest.fn()} />);
    wrapper.find('[name="color"]').at(0).simulate('change', {
      target: { name: 'color', value: '#FF0000' }
    });

    expect(wrapper.state('color')).toEqual('#FF0000');
  });

  it('handles submitting a new color', () => {
    const submit = jest.fn();
    const wrapper = shallow(<CreateColor addColor={submit} />);

    wrapper.find('form').at(0).simulate('submit', {
      preventDefault: jest.fn(),
    });
    
    expect(submit).toHaveBeenCalledWith({ name: wrapper.state('name'), hex: wrapper.state('color') });
    
  });
  

});

