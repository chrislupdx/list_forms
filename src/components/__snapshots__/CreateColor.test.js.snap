// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createcolor component renders a createcolor 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="#00000"
  />
  <button>
    Add Color
  </button>
</form>
`;
